### PR TITLE
FontSizePicker: Fix header order in RTL languages

### PR DESCRIPTION
**Pull request by noisysocks**
## What?
Addresses a bug noticed by @ciampo in https://github.com/WordPress/gutenberg/pull/44483#pullrequestreview-1123284658 where the _Size_ label and hint appear in the wrong order in RTL languages.

## How?
Setting `display: inline-block` makes the browser re-order the elements automatically because `direction: rtl` will apply to all non `inline` elements. Setting `margin-right` instead of `margin-left` when the language is RTL ensures that the margin goes on the correct side of the hint.

## Testing Instructions
Best to test this in the real app as the RTL setting in Storybook doesn't affect the result of `wp.i18n.isRTL()`.

1. Set your language to e.g. Hebrew or Arabic.
2. Navigate to Appearance → Editor → Global Styles → Typography → Paragraph.
3. The hint should appear to the left of the _Size_ label.

## Screenshots or screencast 
| Before | After |
| - | - |
| Screen Shot 2022-09-30 at 11 34 58 | Screen Shot 2022-09-30 at 11 35 06 |

### CODE REVIEW

File: packages/components/src/font-size-picker/styles.ts [hunk 1]
```diff
@@ -19,12 +19,14 @@ export const Container = styled.fieldset`
...
 export const HeaderLabel = styled( BaseControl.VisualLabel )`
+	display: inline-block;
 	margin-bottom: 0;
 `;
 
 export const HeaderHint = styled.span`
 	color: ${ COLORS.gray[ 700 ] };
-	margin-left: ${ space( 1 ) };
+	display: inline-block;
+	${ rtl( { marginLeft: space( 1 ) }, { marginRight: space( 1 ) } )() }
```
Comment (ciampo) [inline packages/components/src/font-size-picker/styles.ts]: _Moar automagic_ in case we kept using this approach: the `rtl` function [should be able](https://github.com/WordPress/gutenberg/blob/44b9b59518a4f04ee1b29e0c1c75b8ffe3e7faa2/packages/components/src/utils/rtl.js#L16-L65) to automatically turn `marginLeft` into `marginRight` !

```suggestion
	${ rtl( { marginLeft: space( 1 ) } )() }
```

**Review by ciampo — COMMENTED** [review]
Thank you @noisysocks for this fix!

Another way to test for these changes is directly in Storybook — we have added a LTR/RTL switcher, so that you don't necessarily have to change the WordPress language settings :) 

https://user-images.githubusercontent.com/1083581/193225212-1d78bab0-04a1-4c6a-8aa3-31c6bdca2ada.mp4

In this case, I noticed that the RTL styles are not applied when the component dynamically updates from LRT to RTL at runtime.
I think this is due some how the fact that a change in writing direction doesn't force the `styled` component to re-render (and therefore recompute styles). One approach could be to use `rtl.watch()` and trigger a re-render, but for such a simple scenario probably the better solution is to use a custom flexbox layout, which completely avoids the need for the `rtl` function, and therefore should also update dynamically when the writing direction changes at runtime!

<details>

<summary>Something like this maybe?</summary>

```diff
diff --git a/packages/components/src/font-size-picker/styles.ts b/packages/components/src/font-size-picker/styles.ts
index 86cdbc08c6..afcdde1b29 100644
--- a/packages/components/src/font-size-picker/styles.ts
+++ b/packages/components/src/font-size-picker/styles.ts
@@ -19,14 +19,14 @@ export const Container = styled.fieldset`
 `;
 
 export const HeaderLabel = styled( BaseControl.VisualLabel )`
-	display: inline-block;
+	display: flex;
+	gap: ${ space( 1 ) };
+	justify-content: flex-start;
 	margin-bottom: 0;
 `;
 
 export const HeaderHint = styled.span`
 	color: ${ COLORS.gray[ 700 ] };
-	display: inline-block;
-	${ rtl( { marginLeft: space( 1 ) }, { marginRight: space( 1 ) } )() }
 `;
 
 export const Controls = styled.div< {
```
</details>

File: packages/components/src/font-size-picker/styles.ts (hunk 1 above)
Comment (noisysocks) [reply packages/components/src/font-size-picker/styles.ts]: Neat!

**Comment by noisysocks** [comment]
Hey @ciampo. I applied your suggestion—works a treat! 🙂

File: .github/dependabot.yml [hunk 2]
```diff
@@ -2,9 +2,9 @@
...
-    directory: "/"
-    schedule:
-      interval: "daily"
-    open-pull-requests-limit: 10
+    # Check for updates to GitHub Actions.
+    - package-ecosystem: 'github-actions'
+      directory: '/'
+      schedule:
+          interval: 'daily'
+      open-pull-requests-limit: 10
```
Comment (ciampo) [inline .github/dependabot.yml]: Are these changes necessary or are they an auto-format leftover?

**Review by ciampo — APPROVED** [review]
Neat 🚀 

Thank you!

File: .github/dependabot.yml (hunk 2 above)
Comment (noisysocks) [reply .github/dependabot.yml]: Oh, not sure where this came from! I'll reset the file.
